sqlmesh/utils/file.py

```diff
@@ -0,0 +1,129 @@
+import json
+import random
+import time
+import typing as t
+from contextlib import contextmanager
+from uuid import uuid4
+
+import fsspec
+
+class FileTransactionHandler:
+    """A transaction handler which wraps a file path."""
+
+    def __init__(self, path: str) -> None:
+        """Creates a new FileTransactionHandler.
+        
+        Args:
+            path: The path to lock. It should follow the format of a table name.
+                Something like snapshots/version=<ver>/<snapshot_id>
+        """
+        # TODO: pass in a filesystem, naturally
+        self._fs: fsspec.AbstractFileSystem = fsspec.filesystem("file")
+        self._path = path
+        self._original_contents: t.Optional[bytes] = None
+        self._is_locked = False
+
+    @property
+    def path(self) -> str:
```

Comment:
delete this and change the property to self.path = path, the getter is not necessary